src/components/EditDeviceForm.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+
+import type { Device, DeviceMutator } from 'brewskey.js-api';
+import type { FormChildProps } from '../common/form/types';
+
+import * as React from 'react';
+// todo replace with scroll view
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
+import { Button } from 'react-native-elements';
+import DeviceBaseFormFields from './DeviceBaseFormFields';
+import Form from '../common/form/Form';
+import FormField from '../common/form/FormField';
+
+type Props = {|
+  device: Device,
+  onSubmit: (values: DeviceMutator) => Promise<void>,
+|};
+
+const EditDeviceForm = ({ device, onSubmit }: Props): React.Node => (
+  <Form>
+    {({ handleSubmit }: FormChildProps): React.Node => (
+      <KeyboardAwareScrollView>
+        <DeviceBaseFormFields device={device} />
+        <FormField initialValue={device.particleId} name="particleId" />
+        <Button
+          title="Edit Brewskey box"
+          onPress={(): Promise<void> => handleSubmit(onSubmit)}
```

Comment:
So I asked internally about how the stateless components work. If you have any event handlers in them you should just use a standard component. Otherwise you're rebinding the `onPress` with every render.